### ISSUE TITLE
bug: can't provide `on present` event handlers on IonModal due to incorrect interface

### ISSUE THREAD

**Issue by alekseykarpenko** [ionic-team/ionic#20249]
# Bug Report

**Ionic version:**

[x] **4.11.8**

**Current behavior:**
I can't apply any of described in [docs](https://ionicframework.com/docs/api/modal) event handlers to `IonModal`(both *dismiss* `onIonModalDidDismiss`/`onIonModalWillDismiss` and *present* `onIonModalDidPresent`/`onIonModalWillPresent`):
![image](https://user-images.githubusercontent.com/4648983/72749908-0020cd80-3bc4-11ea-8eb6-f3b50e0bbf7d.png)

**Expected behavior:**

I want to use mentioned handlers on `IonModal` as described in docs.

**Steps to reproduce:**

Use an example from `IonModal` docs, add handlers for *present* events.
Looks like there is a mistake in interfaces. As I could previously provide `onDidDismiss`/`onWillDismiss`, i had tried also `onDidPresent` and `onWillPresent`, but those also won't work.

**Related code:**

```tsx
import React, { useState } from 'react';
import { IonModal, IonButton, IonContent } from '@ionic/react';

export const ModalExample: React.FC = () => {
  const [showModal, setShowModal] = useState(false);

  return (
    <IonContent>
      <IonModal isOpen={showModal} onIonDidDismiss={() => setShowModal(false)} onIonWillPresent={() => console.log("modal will present")}>
        <p>This is modal content</p>
        <IonButton onClick={() => setShowModal(false)}>Close Modal</IonButton>
      </IonModal>
      <IonButton onClick={() => setShowModal(true)}>Show Modal</IonButton>
    </IonContent>
  );
};
```

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 5.4.11 (/Users/ak/.nvm/versions/node/v10.16.3/lib/node_modules/ionic)
   Ionic Framework : @ionic/react 4.11.8

Capacitor:

   Capacitor CLI   : 1.3.0
   @capacitor/core : 1.3.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v10.16.3 (/Users/ak/.nvm/versions/node/v10.16.3/bin/node)
   npm    : 6.13.6
   OS     : macOS Catalina

```

**Comment by alekseykarpenko** [comment]
⚠️ Looks like duplicate of #19923, sorry.

**Comment by alekseykarpenko** [comment]
Let it be opened for some time to pull attention to the problem

**Comment by elylucas** [comment]
closing as this is a duplicate of #19923 and will be fixed in 4.11.9 release
